gapic-common/lib/gapic/paged_enumerable.rb

```diff
@@ -124,9 +125,10 @@ def next_page
 
       next_request = @request.dup
       next_request.page_token = @page.next_page_token
-      next_response = @grpc_stub.call_rpc @method_name, next_request, options: @options
```

Comment:
I would create a new operation callback that invokes the user callback if provided, and captures the operation object to it can be stored on the Page:

```ruby
next_operation = nil
next_operation_callback = ->(next_resp, next_op) do
  # call the user-supplied callback if present.
  operation_callback.call next_resp, next_op if @operation_callback
  # save the operation to next_operation so we can use it.
  next_operation = next_op
end
next_response = @grpc_stub.call_rpc @method_name, next_request, options: @options
next_response = @grpc_stub.call_rpc @method_name, next_request, options: @options, operation_callback: next_operation_callback

@page = Page.new next_response, next_operation, @resource_field
```

Reply:
how's this?